googleapiclient/discovery.py

```diff
@@ -774,9 +774,21 @@ def method(self, **kwargs):
     if media_filename:
       # Ensure we end up with a valid MediaUpload object.
       if isinstance(media_filename, six.string_types):
-        (media_mime_type, encoding) = mimetypes.guess_type(media_filename)
+        # FIXME: get this from kwargs['media_mime_type'], rather than mucking with body
```

Comment:
Why not go ahead and add a kwarg and fallback to `guess_type` if it's not specified (and issue a warning)?